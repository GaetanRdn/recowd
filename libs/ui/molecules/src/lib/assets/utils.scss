@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
