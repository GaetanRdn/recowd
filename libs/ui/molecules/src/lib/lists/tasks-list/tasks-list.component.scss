:host {
  .rc-list-item {
    &::before {
      content: '✔️ ';
    }

    &:first-letter {
      text-transform: capitalize;
    }
  }
}
