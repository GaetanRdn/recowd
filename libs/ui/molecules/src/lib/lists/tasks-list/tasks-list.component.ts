import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ListModule } from '@recowd/ui-atoms';
import { TranslateModule } from '@ngx-translate/core';

@Component({
  selector: 'rc-tasks-list',
  standalone: true,
  imports: [CommonModule, ListModule, TranslateModule],
  template: ` <rc-list>
    <rc-list-title>{{ 'completed-tasks' | translate }}</rc-list-title>
    <rc-list-item *ngFor="let task of tasks">{{ task }}</rc-list-item>
  </rc-list>`,
  styleUrls: ['tasks-list.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class TasksListComponent {
  @Input() public tasks: string[] = [];
}
