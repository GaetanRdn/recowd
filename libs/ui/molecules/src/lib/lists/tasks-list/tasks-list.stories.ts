import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { TasksListComponent } from './tasks-list.component';

export default {
  title: 'molecules/lists/tasks-list',
  decorators: [
    moduleMetadata({
      imports: [TasksListComponent],
    }),
  ],
} as Meta<TasksListComponent>;

const Template: Story<TasksListComponent> = (args: TasksListComponent) => ({
  props: {
    tasks: args.tasks,
  },
  template: `<rc-tasks-list [tasks]="tasks"></rc-tasks-list>`,
});

export const Default = Template.bind({});
Default.args = {
  tasks: ['few things', 'many things', 'too many things'],
};
