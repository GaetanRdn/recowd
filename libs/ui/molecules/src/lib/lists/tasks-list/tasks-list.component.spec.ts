import { TestBed, waitForAsync } from '@angular/core/testing';
import { TasksListComponent } from './tasks-list.component';
import { TemplateLookup } from '@recowd/test/utils';
import { Component } from '@angular/core';
import { TranslateModule } from '@ngx-translate/core';

describe('TasksListComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [TasksListComponent, TranslateModule.forRoot()],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
    templateLookup.detectChanges();
  }));

  it('should create empty', () =>
    expect(templateLookup.firstChildElement).toMatchSnapshot());

  it('should create with technos', () => {
    templateLookup.hostComponent.tasks = [
      'few things',
      'many things',
      'too many things',
    ];

    templateLookup.detectChanges();

    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });
});

@Component({
  template: ` <rc-tasks-list [tasks]="tasks"></rc-tasks-list>`,
})
class HostComponent {
  public tasks: TasksListComponent['tasks'] = [];
}
