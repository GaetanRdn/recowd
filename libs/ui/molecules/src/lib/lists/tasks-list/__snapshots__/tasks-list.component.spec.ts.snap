// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksListComponent should create empty 1`] = `
<rc-tasks-list>
  <rc-list
    role="list"
  >
    <rc-list-title
      class="rc-list-title"
    >
      completed-tasks
    </rc-list-title>
    <span
      class="rc-list-items-wrapper"
    >
      
    </span>
  </rc-list>
</rc-tasks-list>
`;

exports[`TasksListComponent should create with technos 1`] = `
<rc-tasks-list>
  <rc-list
    role="list"
  >
    <rc-list-title
      class="rc-list-title"
    >
      completed-tasks
    </rc-list-title>
    <span
      class="rc-list-items-wrapper"
    >
      <rc-list-item
        class="rc-list-item"
        role="list-item"
      >
        few things
      </rc-list-item>
      <rc-list-item
        class="rc-list-item"
        role="list-item"
      >
        many things
      </rc-list-item>
      <rc-list-item
        class="rc-list-item"
        role="list-item"
      >
        too many things
      </rc-list-item>
      
    </span>
  </rc-list>
</rc-tasks-list>
`;
